jwt/src/main/java/me/uport/sdk/jwt/JWTTools.kt

```diff
@@ -227,20 +231,29 @@ class JWTTools(
 
         val publicKeys = resolveAuthenticator(header.alg, payload.iss, auth)
 
-        val signingInputBytes = token.substringBeforeLast('.').toByteArray()
+        val signingInputBytes = token.substringBeforeLast('.').toByteArray(utf8)
 
         val sigData = signatureBytes.decodeJose()
 
-        if (header.alg == JwtHeader.ES256K_R) {
-            if (verifyRecoverableES256K(publicKeys, sigData, signingInputBytes)) return payload
-        }
-        else if (header.alg == JwtHeader.ES256K) {
-            if (verifyES256K(publicKeys, sigData, signingInputBytes)) return payload
+        val signatureIsValid = verificationMethod[header.alg]
+                ?.invoke(publicKeys, sigData, signingInputBytes)
+                ?: throw InvalidAlgorithmParameterException("JWT algorithm ${header.alg} not supported")
```

Comment:
This looks good

Reply:
It's still a guess how far down the kotlin rabbit hole to go. This seemed like a good compromise between readability and extensibility.
There's very little code repetition and the if/else tree won't grow as we add new algorithms support. Plus, the algorithms and the actual implementation are now better tied together so there's less room for mistakes.